### PR TITLE
add caravel support

### PR DESCRIPTION
**Pull request by rocky1001**
transform label name to lower letters, which can be loaded by caravel(pandas),

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -4,24 +4,23 @@ Kylin DBAPI Driver and Sqlalchemy Dialect
...
+    $ cd pykylin
+    $ pip install .
 
 
 More info
 ---------
 
- * http://kylin.incubator.apache.org/
- * http://www.sqlalchemy.org/
+ * forked from https://github.com/wxiang7/pykylin Thanks origin author: @wxiang7
```
Comment (wxiang7) [inline README.rst:17]: Move revision info to a new file named with "CHANGES.rst"

File: pykylin/cursor.py [hunk 2]
```diff
@@ -17,15 +24,51 @@ def __init__(self, connection):
...
 
     def callproc(self):
-        raise('Stored procedures not supported in Kylin')
+        raise ('Stored procedures not supported in Kylin')
 
     def close(self):
         logger.debug('Cursor close called')
 
+    @staticmethod
+    def trans_sql_for_kylin(sql_str):
```
Comment (wxiang7) [inline pykylin/cursor.py:33]: What's the purpose of transforming sql query to another version ?

This is not what a Cursor instance should do. If SQL syntax for Kylin is different from other SQL dialects, you should consider change "dialect.py".

FYI:
[SQLAlchemy Dialect](https://github.com/zzzeek/sqlalchemy/blob/master/README.dialects.rst)

File: pykylin/cursor.py [hunk 3]
```diff
@@ -17,15 +24,51 @@ def __init__(self, connection):
...
+        sql_str = sql_str.lower()
```
Comment (wxiang7) [inline pykylin/cursor.py]: Kylin syntax is always case insensitive ?

File: pykylin/cursor.py [hunk 4]
```diff
@@ -17,15 +24,51 @@ def __init__(self, connection):
...
+        if not rep_sql_list:
+            return sql_str
+        transformed_sql = ''
+        rep_sql_dict = dict()
+        rep_sql_set = set()
+        for rep_sql in rep_sql_list:
+            rep_sql_dict[rep_sql] = rep_sql.split('|')[-2]
+            rep_sql_set.add(rep_sql.split('|')[1])
+
+        # step 1, replace all the const str to data name
```
Comment (wxiang7) [inline pykylin/cursor.py:45]: These hand-written transforming steps tends to be very tricky.
SQLAlchemy's visitor pattern could probably help you.

File: pykylin/cursor.py [hunk 5]
```diff
@@ -57,11 +100,13 @@ def _type_mapped(self, result):
             if tpe == 'DATE':
                 val = parser.parse(val)
             elif tpe == 'BIGINT' or tpe == 'INT' or tpe == 'TINYINT':
-                val = int(val)
+                val = int(val) if val else 0
```
Comment (wxiang7) [inline pykylin/cursor.py:108]: We need to keep the "null" information. How about

```
val = int(val) if val else None
```

File: README.rst (hunk 1 above)
Comment (rocky1001) [inline README.rst:17]: Got it, not so familiar with github fork conventions, sorry.

File: pykylin/cursor.py (hunk 2 above)
Comment (rocky1001) [inline pykylin/cursor.py:33]: All these tricky sql transform steps are because Caravel does't support multi-table join directly.
So we put the multi-table join transform into pykylin as a simple work around.
We have plan to modify Caravel for multi-table join support. The code here will be deleted in the future.

File: pykylin/cursor.py (hunk 5 above)
Comment (rocky1001) [inline pykylin/cursor.py:108]: That seems OK to me, but we have to add some test cases to testify the None value in Caravel.
